hyperspy/component.py

```diff
@@ -756,6 +756,9 @@ def summary(self):
                                                parameter.std,
                                                parameter.units)
 
+    def __tempcall2d__(self, p, x, y, onlyfree=True):
+        self.fetch_values_from_array(p, onlyfree=onlyfree)
+        return self.function(x, y)
```

Comment:
I think we could get away with not having to define a 2D version of this if we simply replace `x` in the original with `position`, and pass a tuple `(x, y)` for anything 2D. `Gaussian2D.function()` would of course have to be modified to take a tuple instead of two separate arguments, but that's no problem. Should give better code reusability, meaning any changes to `__tempcall__` won't have to be mirrored in 2D version, and it will also be able to support 3D (or higher) versions later on at no extra cost.